common/notification/index.ts

```diff
@@ -261,16 +262,39 @@ async function deliverNotification(
 
     try {
         const user = getUserForTypeORM(legacyUser);
-        const channelsToSendTo = channels.filter((it) => it.canSend(notification, user));
-        if (!channelsToSendTo || channelsToSendTo.length === 0) {
-            throw new Error(`No fitting channel found for Notification(${notification.id})`);
-        }
 
         if (notification.hookID) {
             await triggerHook(notification.hookID, user);
         }
 
+        const { messageType } = getMessage(concreteNotification);
+
+        // default channel is inapp
+        const activeChannelForNotificationType: string[] = ['inapp'];
+
         // TODO: Check if user silenced this notification
+        const { notificationPreferences } = await getStudentOrPupil(user);
+
+        if (notificationPreferences && typeof notificationPreferences === 'string') {
+            const enabledChannels: { [channel: string]: boolean } = JSON.parse(notificationPreferences)[messageType];
+
+            for (const channel in enabledChannels) {
+                if (enabledChannels[channel] === true) {
+                    activeChannelForNotificationType.push(channel);
```

Comment:
I don't see the point of `activeChannelForNotificationType`, what benefit does this array have over the `enabledChannels` dictionary?